sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/Http2ConnectionConfig.java

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.Configs;
+import com.azure.cosmos.util.Beta;
+
+@Beta(value = Beta.SinceVersion.V4_66_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+public class Http2ConnectionConfig {
+    private static final int DEFAULT_MAX_CONCURRENT_STREAMS = 30;
+    private static final int DEFAULT_MIN_CONNECTION_POOL_SIZE = 1;
+
+    private int maxConnectionPoolSize;
+    private int minConnectionPoolSize;
+    private int maxConcurrentStreams;
+    private boolean enabled;
+
+    @Beta(value = Beta.SinceVersion.V4_66_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public Http2ConnectionConfig() {
+        this.maxConnectionPoolSize = Configs.getDefaultHttpPoolSize(); // overlapping with the maxConnectionPoolSize in gateway connection config
+        this.minConnectionPoolSize = DEFAULT_MIN_CONNECTION_POOL_SIZE;
```

Comment:
Shouldn't the default values be populated from the new env variables/Config?

Reply:
this.maxConnectionPoolSize = Configs.getHttp2MaxConnectionPoolSize();
        this.minConnectionPoolSize = Configs.getHttp2MinConnectionPoolSize();
        this.maxConcurrentStreams = Configs.getHttp2MaxConcurrentStreams();
        this.enabled = Configs.isHttp2Enabled();

Reply:
Good catch, forgot to wire them up when switching, updated